lib/sync.js

```diff
@@ -0,0 +1,216 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const utils = require('./utils');
+const noop = value => () => value;
+
+const kPath = Symbol('path');
+const kRealPathExists = Symbol('realpath-exists');
+const kResult = Symbol('result');
+
+const defaults = {
+  depth: Infinity,
+  objects: false,
+  realpath: false,
+  recursive: false,
+
+  onDirectory: null,
+  onEach: null,
+  onEmpty: null,
+  onFile: null
+};
+
+const readdirSync = (basedir, options = {}) => {
+  if (Array.isArray(basedir)) {
+    return readdirsSync(basedir, options);
+  }
+
+  const opts = { ...defaults, ...options };
+
+  const isMaxDepth = file => file.depth >= opts.depth;
+  const isIgnoredFile = opts.isIgnoredFile || noop(false);
+  const isIgnoredDir = opts.isIgnoredDir || noop(false);
+  const isMatch = opts.isMatch ? utils.matcher(opts.isMatch, opts) : () => true;
```

Comment:
Same suggestion as in `async`:

```suggestion
  const isMatch = opts.isMatch ? utils.matcher(opts.isMatch, opts) : noop(true);
```